mne/io/snirf/tests/test_snirf.py

```diff
@@ -413,25 +443,80 @@ def test_snirf_fieldtrip_od():
 
 
 @requires_testing_data
-def test_snirf_kernel_hb():
-    """Test reading Kernel SNIRF files with haemoglobin data."""
-    raw = read_raw_snirf(kernel_hb, preload=True)
-
-    # Test data import
-    assert raw._data.shape == (180 * 2, 14)
-    assert raw.copy().pick("hbo")._data.shape == (180, 14)
-    assert raw.copy().pick("hbr")._data.shape == (180, 14)
-
-    assert_allclose(raw.info["sfreq"], 8.257638)
+@pytest.mark.parametrize(
+    "kind, ver, shape, n_nan, fname",
+    [
+        pytest.param("hb", "new", (4, 38), 0, kernel_hb, id="hb"),
+        pytest.param("hb", "old", (180 * 2, 14), 20, kernel_hb_old, id="hb old"),
+        pytest.param(
+            "td moments", "new", (12, 38), 0, kernel_td_moments, id="td moments"
+        ),
+        pytest.param("td gated", "new", (100, 38), 0, kernel_td_gated, id="td gated"),
+        pytest.param(
+            "td moments",
+            "old",
+            (1080, 14),
+            60,
+            kernel_td_moments_old,
+            id="td moments old",
+        ),
+    ],
+)
+def test_snirf_kernel_basic(kind, ver, shape, n_nan, fname):
+    """Test reading Kernel SNIRF files with haemoglobin or TD data."""
+    raw = read_raw_snirf(fname, preload=True)
+    if kind == "hb":
+        # Test data import
+        assert raw._data.shape == shape
+        hbo_data = raw.get_data("hbo")
+        hbr_data = raw.get_data("hbr")
+        assert hbo_data.shape == hbr_data.shape == (shape[0] // 2, shape[1])
+        hbo_norm = np.nanmedian(np.linalg.norm(hbo_data, axis=-1))
+        hbr_norm = np.nanmedian(np.linalg.norm(hbr_data, axis=-1))
+        # TODO: Old file vs new file scaling, one is wrong!
```

Comment:
Would be good to know if the numbers for the newer one are correct or not...

Reply:
The way the code works now the old test file and the new test file differ by several orders of magnitude (like 1e5) so I think there is uM vs M error scaling either in the file or in the code. Based on It looks like they should be closer to ~1e-6, right? If so then there is some bug in how we're reading the hbo/hbr values for the old file